File: Vehicle.java
public class Vehicle {
  //properties
  // public Vehicle(String color, int wheels) {
  //   this.color = color;
  //   this.wheels = wheels;
  // }
  private String color;
  private int wheels;
  //getters
  public String getColor() {
    return this.color;
  }
  public int getWheels() {
    return this.wheels;
  }
  //setters
  public void setColor(String color) {
    this.color = color;
  }
  public void setWheels(int wheels) {
    this.wheels = wheels;
  }
  //methods
  public String honk() {
    return "Honk!";
  }
  public String output() {
    return "This vehicle has " + this.wheels + " wheels and is " + this.color + ".";
  }
  public static void main(String [] args) {
    Vehicle herbie = new Vehicle();

    System.out.println(herbie.honk());
  }
}

File: vehicle.js
var Vehicle = function(color, wheels) {
  this.color = color;
  this.wheels = wheels;
}

Vehicle.prototype.output = function() {
  return "This vehicle has " + this.wheels + " wheels and is " + this.color + ".";
}

var cliff = new Vehicle("red", 4);

// console.log(cliff.output());

var Car = function(color) {
  this.wheels = 4;
  this.color = color;
}

Car.prototype = new Vehicle();

Car.prototype.drive = function(miles) {
  return 'This car just drove ' + miles
 + ' miles.';
}

var herbie = new Car('white');
console.log(herbie.output());
console.log(herbie.drive(200));